command/peering/export/export.go

```diff
@@ -0,0 +1,187 @@
+		return 1
+	}
+
+	entry, _, err := client.ConfigEntries().Get("exported-services", "default", nil)
+	if err != nil && !strings.Contains(err.Error(), agent.ConfigEntryNotFoundErr) {
+		c.UI.Error(fmt.Sprintf("Error reading config entry %s/%s: %v", "exported-services", "default", err))
+		return 1
+	}
+	if entry == nil {
+		cfg := api.ExportedServicesConfigEntry{
+			Name: "default",
+			Services: []api.ExportedService{
+				{
+					Name:      c.serviceName,
+					Namespace: c.serviceNamespace,
+					Consumers: buildConsumersFromPeerNames(peerNames),
+				},
+			},
+		}
+
+		_, _, err = client.ConfigEntries().Set(&cfg, &api.WriteOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating config entry: %s", err))
+			return 1
+		}
+
+	} else {
+
+		cfg, ok := entry.(*api.ExportedServicesConfigEntry)
+		if !ok {
+			c.UI.Error(fmt.Sprintf("Existing config entry has incorrect type: %t", entry))
+			return 1
+		}
+
+		serviceExists := false
+
+		for i, service := range cfg.Services {
+
+			if service.Name == c.serviceName && service.Namespace == c.serviceNamespace {
+
+				serviceExists = true
+				for _, peerName := range peerNames {
+					peerExists := false
+
+					for _, consumer := range service.Consumers {
+						if consumer.Peer == peerName {
+							peerExists = true
+							break
+						}
+					}
+
+					if !peerExists {
+						cfg.Services[i].Consumers = append(cfg.Services[i].Consumers, api.ServiceConsumer{Peer: peerName})
+					}
+				}
+			}
+		}
+
+		if !serviceExists {
+			cfg.Services = append(cfg.Services, api.ExportedService{
+				Name:      c.serviceName,
+				Namespace: c.serviceNamespace,
+				Consumers: buildConsumersFromPeerNames(peerNames),
+			})
+		}
+
+		ok, _, err := client.ConfigEntries().CAS(cfg, cfg.GetModifyIndex(), nil)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error writing modifed service and peer to config entry: %s", err))
+			return 1
+		}
+
+		if !ok {
+			c.UI.Error(fmt.Sprintf("Modifed service and peer config entry was changed during update. Please try again"))
+			return 1
+		}
+	}
+	c.UI.Info(fmt.Sprintf("Successfully exported service %s to peers %s", c.serviceName, c.peerNames))
+	return 0
+}
+
+func buildConsumersFromPeerNames(peerNames []string) []api.ServiceConsumer {
+	consumers := []api.ServiceConsumer{}
+	for _, peer := range peerNames {
+		consumers = append(consumers, api.ServiceConsumer{
+			Peer: peer,
+		})
+	}
+	return consumers
+}
+
+//========
+
+func (c *cmd) Synopsis() string {
+	return synopsis
+}
+
+func (c *cmd) Help() string {
+	return flags.Usage(c.help, nil)
+}
+
+const (
+	synopsis = "Export a service"
+	help     = `
+Usage: consul peering export [options] -service <service name> -peers <other cluster name>
+
+  Export a service. The peers provided will be used locally by
+  this cluster to refer to the other cluster where the services will be exported. 
+
+  Example:
+
+	$ consul peering export -service=web -peers=other-cluster
```

Comment:
Might be worth including multiple peers in the example, `-peers=cluster-1,cluster-2`